service/storage_service.cc

```diff
@@ -1967,8 +1967,16 @@ class topology_coordinator : public endpoint_lifecycle_subscriber {
             if (holder && holder->failed()) {
                 holder->ignore_ready_future(); // Prevent warnings in logs
             }
-            holder = futurize_invoke(action)
-                        .finally([this, g = _async_gate.hold(), gid, name] () noexcept {
+            holder = futurize_invoke(action).then_wrapped([this, gid, name] (future<> f) {
+                if (f.failed()) {
+                    auto ep = f.get_exception();
+                    slogger.error("raft topology: {} for tablet {} failed: {}", name, gid, ep);
+                    return seastar::sleep_abortable(std::chrono::seconds(1), _as).then([ep] () mutable {
```

Comment:
Suggest exponential backoff.

Reply:
100ms -> 4s ?

Reply:
Will it retry forever? How can the user abort the ops? After the abort, will it rollback to a previous state as if removenode/replace did not happen?

Reply:
It will stop retrying when the operation eventually succeeds.

There is no support for automatic abort.

There will be rollback if the involved node is dead and admin marks it as excluded: https://github.com/scylladb/scylladb/issues/16527